### PR TITLE
[Tags] Adding detailed params to tags

### PR DESCRIPTION
**Pull request by jgchoppe**
Add a new query params "detailed" which is a boolean, if detailed is true, you'll have all the usage types for an AWS product.

### CODE REVIEW

File: costs/tags/tags_values.go [hunk 1]
```diff
@@ -92,10 +112,54 @@ func getTagsValuesWithParsedParams(ctx context.Context, params tagsValuesQueryPa
 		return returnCode, errors.GetErrorMessage(ctx, err)
 	}
 	err = json.Unmarshal(*res.Aggregations["data"], &typedDocument)
+	l.Debug("typedDocument ========", map[string]interface{}{
```
Comment (hug33k) [inline costs/tags/tags_values.go]: This logging should not be there

File: costs/tags/tags_values.go [hunk 2]
```diff
@@ -92,10 +112,54 @@ func getTagsValuesWithParsedParams(ctx context.Context, params tagsValuesQueryPa
...
+}
+
+//getTagsResponseDetailed get response for tagging when detailed is true
+func getTagsResponseDetailed(typedDocument esTagsValuesDetailedResult, params tagsValuesQueryParams) TagsValuesResponse {
+	response := TagsValuesResponse{}
+	for _, key := range typedDocument.Keys.Buckets {
+		var values []TagsValues
+		for _, tag := range key.Tags.Buckets {
+			var products []TagValueDetailed
+			var valueDetailed []ValueDetailed
```
Comment (hug33k) [inline costs/tags/tags_values.go]: Wrong place for this declaration

**Review by hug33k — CHANGES_REQUESTED** [review]

File: costs/tags/tags_values.go (hunk 1 above)
Comment (jgchoppe) [reply costs/tags/tags_values.go]: Fixed

File: costs/tags/tags_values.go (hunk 2 above)
Comment (jgchoppe) [reply costs/tags/tags_values.go]: Fixed

File: costs/tags/tags_values.go [hunk 3]
```diff
@@ -69,10 +78,21 @@ type (
...
+		UsageTypes []ValueDetailed `json:"usagetypes"`
+	}
+
 	// contain a tag and the list of products associated
 	TagsValues struct {
-		Tag   string     `json:"tag"`
-		Costs []TagValue `json:"costs"`
+		Tag       string             `json:"tag"`
+		Costs     []TagValue         `json:"costs,omitempty"`
+		Items     []TagValueDetailed `json:"items,omitempty"`
```
Comment (cruffinoni) [inline costs/tags/tags_values.go]: Trailing spaces after variable name declaration.

File: costs/tags/tags_values.go [hunk 4]
```diff
@@ -34,32 +34,41 @@ type (
 		Type   string
 	}
 
+
 	// structs that allows to parse ES result
-	esTagsValuesResult struct {
+	esTagsValuesDetailedResult struct {
 		Keys struct {
 			Buckets []struct {
 				Key  string       `json:"key"`
```
Comment (cruffinoni) [inline costs/tags/tags_values.go]: Missing spaces after variable type declaration.

**Review by cruffinoni — CHANGES_REQUESTED** [review]
Missing/trailing spaces are needed/present.

File: costs/tags/tags_values.go (hunk 3 above)
Comment (jgchoppe) [reply costs/tags/tags_values.go]: Fixed

**Review by hug33k — APPROVED** [review]

**Review by cruffinoni — APPROVED** [review]

**Review by giubil — APPROVED** [review]

**Pull request merged**